src/lib/Generator/File/ExportFileGenerator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\PersonalizationClient\Generator\File;
+
+use EzSystems\EzPlatformRest\Output\Generator;
+use EzSystems\EzRecommendationClient\File\FileManagerInterface;
+use Ibexa\PersonalizationClient\Generator\ItemList\ItemListElementGeneratorInterface;
+use Ibexa\PersonalizationClient\Value\Export\FileSettings;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+final class ExportFileGenerator implements ExportFileGeneratorInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private FileManagerInterface $fileManager;
+
+    private ItemListElementGeneratorInterface $itemListElementGenerator;
+
+    private Generator $outputGenerator;
+
+    public function __construct(
+        FileManagerInterface $fileManager,
+        ItemListElementGeneratorInterface $itemListElementGenerator,
+        Generator $outputGenerator,
+        ?LoggerInterface $logger = null
+    ) {
+        $this->fileManager = $fileManager;
+        $this->itemListElementGenerator = $itemListElementGenerator;
+        $this->outputGenerator = $outputGenerator;
+        $this->logger = $logger ?? new NullLogger();
+    }
+
+    public function generate(FileSettings $fileSettings): void
+    {
+        $this->outputGenerator->reset();
+        $this->outputGenerator->startDocument($fileSettings->getItemList());
+
+        $itemList = $fileSettings->getItemList();
+        $this->itemListElementGenerator->generateElement($this->outputGenerator, $itemList);
+        $filePath = $this->fileManager->getDir() . $fileSettings->getChunkPath();
+        $this->fileManager->save($filePath, $this->outputGenerator->endDocument($itemList));
```

Comment:
This seems to be done a little bit too late. When generating the output for API, files should be already in place in my opinion.

If I understand correctly, the file manager also has two responsibilities: it saves the file and does something with the result of `Generator::endDocument`? Why is closing the document part what it does?